app/code/Magento/Checkout/view/frontend/web/js/view/payment/list.js

```diff
@@ -193,7 +193,7 @@ define([
                 title = this.defaultGroupTitle;
             }
 
-            return title + ':';
```

Comment:
I assume colon should be moved to the template then, to preserve consistency with the current look.